### ISSUE TITLE
Keep BF16 weights in memory as ushort and SIMD-dot directly (skip the load-time F32 widen)

### ISSUE THREAD

**Issue by khurram-uworx** [khurram-uworx/nivara#387]
## Context

Qwen2.5-0.5B-Instruct ships **BF16 on disk** (989 MB, all 290 tensors). The current `qwen` load path (branch `khurram/qwen-inference`, PR #385) reads the raw BF16 payload as `ushort[]` and **widens once to F32 at load** (`ReadUInt16` → SIMD `WidenBf16ToF32` → `float[]`), so ~2 GB of F32 weights then drive a pure-F32 generation loop. The widen costs ~2.2 s once at load and doubles the in-memory footprint vs the on-disk size.

## Proposal

Keep the weights in memory as **raw `ushort` (BF16 bits)** — skip the load-time F32 widen:

- `Vector<ushort>` has **2× the SIMD lanes** of `Vector<float>` (16 vs 8 on AVX2; 32 vs 16 on AVX-512) and moves half the bytes per load.
- In-memory weights halve to ~1 GB (vs ~2 GB F32).
- No load-time widen pass (kills the ~2.2 s startup conversion).

## ⚠️ Critical: integer math is NOT BF16 math

**BF16 bit patterns cannot be multiplied as integers.** `Vector<ushort>` integer `multiply/accumulate` on raw BF16 bits produces numerically **wrong** results — the value's exponent occupies the high 8 bits, so the integer product of two bit patterns does not equal the BF16 product. Any implementation **must not** do arithmetic directly on the raw `ushort` lanes.

## Suggested solution (our approach)

**Widen inside the kernel** — keep `ushort` weights, but widen per-lane on demand with the cheap bit trick and reuse the existing F32 dot kernels:

```
ushort lane w  →  (uint)w << 16   (one lane-wise shift → the F32 bit pattern, in place)
              →  existing F32 dot kernels (TensorPrimitives / SIMD)
```

- Same **widen-compute-narrow** shape as the existing `UseWidenSimd` BF16 kernels (`NivaraPrimitives.UseWidenSimd`, SmolLM's native-BF16 mode); the delta is **raw-ushort storage, wider memory loads, and the shift-based convert**.
- The load-time pass is traded for a per-matmul widen that is nearly free at the lane level.
- Honest expectation: primary win = **memory + memory bandwidth**; decode speed must be **measured**, not assumed — on this CPU pure-F32 in-memory compute measured **~1.5–2× faster than SmolLM's per-matmul BF16 widen** (README narrow-precision section).

## Library support needed (the real ask — core `src/Nivara`)

- A **BF16-weights (raw ushort) × F32-activations matmul** kernel family — forward + VJP (+ JVP) — widening inside the kernel (correct math), `TensorPrimitives`-style span kernels, with a scalar fallback, following the existing kernel conventions (`KernelSelector` dispatch).
- An **opt-in `qwen` flag** to switch between *load-widen* (current default) and *in-memory-ushort* so the two can be A/B'd without changing defaults.
- Tests: parity vs the load-widen path (logits ≤ tolerance, e.g. 1e-4), per repository testing conventions; a README benchmark row (memory + ms/token) in `samples/NivaraInference`.

## Acceptance criteria

- [ ] Correct BF16 math on in-memory ushort weights — parity with the load-widen F32 path (fixture/parity tests).
- [ ] Memory: ~1 GB weight footprint (vs ~2 GB F32) measured in the sample run.
- [ ] A/B `qwen` (load-widen vs in-memory-ushort) median-of-3 decode + load timing; result recorded in the README.
- [ ] Fixture-gated tests silently `Assert.Ignore` when new fixtures are absent (existing convention).
